src/workerd/jsg/resource.h

```diff
@@ -191,6 +214,308 @@ template <typename TypeWrapper,
     typename Indexes>
 struct MethodCallback;
 
+// Helper to determine if a type can be used as a return type in V8 Fast API
+template <typename T>
+struct IsFastApiCompatibleReturnType {
+  static constexpr bool value = std::is_same_v<T, void> || std::is_same_v<T, bool> ||
+      std::is_same_v<T, int32_t> || std::is_same_v<T, uint32_t> || std::is_same_v<T, float> ||
+      std::is_same_v<T, double>;
+};
+
+// Helper to determine if a type can be used as a parameter in V8 Fast API
+template <typename T>
+struct IsFastApiCompatibleParamType {
```

Comment:
If Fast API supports `v8::Local<v8::Value>`, that implies that Fast API actually supports _any_ parameter type. We just have to do the conversion.

Instead of requiring the target function to declare its members as purely numbers or v8::Locals, we should instead say that the target function can use any JSG-compatible type, and we should automatically apply the correct unwrapper to convert a `v8::Local` to the correct type for the function call.

We should, of course, special-case the primitive numeric types, to expose those directly to V8 rather than forcing it to provide us with a handle.

Reply:
Yes, agreed. I'll update this particular part, once I figure out the internals of type wrapper.